ogr/services/gitlab.py

```diff
@@ -112,23 +126,183 @@ def _fork_gracefully(target_repo):
+            status=gitlab_pr.state,
+            url=gitlab_pr.web_url,
+            description=gitlab_pr.description,
+            author=gitlab_pr.author["username"],
+            source_branch=gitlab_pr.source_branch,
+            target_branch=gitlab_pr.target_branch,
+            created=gitlab_pr.created_at,
+        )
+
+    def get_pr_info(self, pr_id: int) -> PullRequest:
+        mr = self.repo.mergerequests.get(pr_id)
+        return self._pr_from_gitlab_object(mr)
+
+    def list_pull_requests(self) -> List[PullRequest]:
+        mrs = self.repo.mergerequests.list(order_by="updated_at", sort="desc")
+        return [self._pr_from_gitlab_object(mr) for mr in mrs]
+
+    def create_pr(
+        self, source_branch: str, target_branch: str, title: str, labels: List[str]
+    ) -> PullRequest:
+        mr = self.repo.mergerequests.create(
             {
-                "id": mr.iid,
-                "title": mr.title,
-                "author": mr.author["username"],
-                "url": mr.web_url,
+                "source_branch": source_branch,
+                "target_branch": target_branch,
+                "title": title,
+                "labels": labels,
             }
-            for mr in mrs
+        )
+        return self._pr_from_gitlab_object(mr)
+
+    def update_pr_info(self, pr_id: int, description: str) -> PullRequest:
+        pr = self.repo.mergerequests.get(pr_id)
+        pr.description = description
+        pr.save()
+        return self._pr_from_gitlab_object(pr)
+
+    def get_all_pr_commits(self, pr_id: int) -> List[str]:
+        mr = self.repo.mergerequests.get(pr_id)
+        return [commit.id for commit in mr.commits()]
+
+    @staticmethod
+    def _prcomment_from_gitlab_object(raw_comment) -> PRComment:
+        return PRComment(
+            comment=raw_comment.body,
+            author=raw_comment.author["username"],
+            created=raw_comment.created_at,
+            edited=raw_comment.updated_at,
+        )
+
+    def _get_all_pr_comments(self, pr_id: int) -> List[PRComment]:
+        pr = self.repo.mergerequests.get(pr_id)
+        return [
+            self._prcomment_from_gitlab_object(raw_comment)
+            for raw_comment in pr.notes.list()
+        ]
+
+    def pr_comment(self, pr_id: int, body: str) -> PRComment:
+        """
+        Create comment on an pr.
+        """
+        pr = self.repo.issues.get(pr_id)
+        comment = pr.notes.create({"body": body})
+        return self._prcomment_from_gitlab_object(comment)
+
+    def _git_tag_from_tag_name(self, tag_name: str) -> GitTag:
+        git_tag = self.repo.tags.get(tag_name)
+        return GitTag(name=git_tag.name, commit_sha=git_tag.commit["id"])
+
+    @staticmethod
+    def _release_from_gitlab_object(raw_release, git_tag: GitTag) -> Release:
+        return Release(
+            title=raw_release.name,
+            body=raw_release.description,
+            tag_name=raw_release.tag_name,
+            url=None,
```

Comment:
It looks like there is no such web-url. There is only a web-url for tags:

Example:
- releases: https://gitlab.gnome.org/World/OpenPaperwork/paperwork/-/releases
- tag: https://gitlab.gnome.org/World/OpenPaperwork/paperwork/-/tags/1.2.4

We can leave it empty or use the tag url.